### ISSUE TITLE
[bug] default_python_requires_id_mode throw an error

### ISSUE THREAD

**Issue by boussaffawalid** [conan-io/conan#10954]
I tried to change `default_python_requires_id_mode` as explained in https://docs.conan.io/en/latest/extending/python_requires.html to

```
[general]
default_python_requires_id_mode = unrelated_mode
```

but I got this exception when creating a package

```
10:36:05  Traceback (most recent call last):
10:36:05    File "C:\Python38\lib\site-packages\conans\model\info.py", line 296, in __init__
10:36:05      func_package_id_mode = getattr(self, default_package_id_mode)
10:36:05  AttributeError: 'PythonRequireInfo' object has no attribute 'unrelated_mode'
```

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows server 2019
  * Compiler+version: msvc 2019
  * Conan version:  1.46.2
  * Python version: 3.8

**Comment by lasote** [comment]
Hi, thanks for reporting. I open a PR solving the issue, unless I'm missing something this is indeed a bug.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10959 fixed it, will be released 1.48. Thanks!
